utils/frame/benchmarking-cli/src/machine/hardware.rs

```diff
@@ -18,8 +18,58 @@
 //! Contains types to define hardware requirements.
 
 use lazy_static::lazy_static;
-use serde::{Deserialize, Serialize};
-use std::fmt;
+use sc_sysinfo::Throughput;
+use serde::{
+	de::{Error, MapAccess, Visitor},
+	ser::SerializeMap,
+	Deserialize, Deserializer, Serialize, Serializer,
+};
+use sp_std::{fmt, fmt::Formatter};
+
+pub fn serialize_throughput<S>(t: &Throughput, serializer: S) -> Result<S::Ok, S::Error>
+where
+	S: Serializer,
+{
+	let mut map = serializer.serialize_map(Some(1))?;
+	let (value, unit) = t.normalize();
+	map.serialize_entry(&unit.to_string(), &value)?;
```

Comment:
As I've said in the previous comment, the problem here was that it was not obvious in the `sysinfo` crate that the exact string returned from `normalize` is used here for serialization/deserialization.

You changed it to use an enum (which is good!), but you still kept the "enum -> serialization key" conversion exactly where it was previously (in the `sysinfo` crate). The whole point of making this into an enum was so that we can move that conversion here so that 1) how it's printed, and 2) how it's serialized here are kept entirely independent. That way if someone were to modify either one of them in the future they won't inadvertently break the other one. No spooky action at a distance.

So this should not use `to_string` on the enum; just do a `match` on the enum instead.

Reply:
It's fine either way. The whole point of `Display` is, well, displaying something to the user, and that's what we're doing in `sc-sysinfo`.

(Note: if something implements `Display` then you don't need to `.to_string()`; you can just print it directly.)

Reply:
Other than the nit you mentioned do the changes look good now?